docs/tutorial/writing-tests/test-case.rst

```diff
@@ -0,0 +1,1105 @@
+to customize testing parameters that have to be defined per-machine/test run.
+Consider the following:
+
+.. code-block:: none
+
+    id: network_speed
+    flags: simple
+    _summary: Test that the network speed is acceptable
+    environ:
+      ACCEPTABLE_BYTES_PER_SECOND_SPEED
+    command:
+      echo Testing for the limit speed: ${ACCEPTABLE_BYTES_PER_SECOND_SPEED:-600}
+      curl -Y ${ACCEPTABLE_BYTES_PER_SECOND_SPEED:-600} -o /dev/null \
+        https://cdimage.ubuntu.com/ubuntu-mini-iso/noble/daily-live/current/noble-mini-iso-amd64.iso
+
+Before running the test we have to define a Checkbox configuration. Note that
+if we were using a test plan, we could run it with a launcher, but the
+``run`` command doesn't take a launcher parameter, so we have to use a
+configuration file. Place the following in ``~/.config/checkbox.conf``.
+
+.. code-block:: ini
+
+    [environment]
+    ACCEPTABLE_BYTES_PER_SECOND_SPEED=60000000
+
+Running the test with the usual command, you will notice that now the limit is
+higher:
+
+.. code-block:: none
+
+    (checkbox_venv) > checkbox-cli run com.canonical.certification::network_speed
+    [...]
+    Testing for the limit speed: 60000000
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  5105    0  5105    0     0   6645      0 --:--:-- --:--:-- --:--:--  6647
+    ------------------------------------------------------------------------- >8 ---
+    Outcome: job passed
+    Finalizing session that hasn't been submitted anywhere: checkbox-run-2024-08-06T14.17.23
+    ==================================[ Results ]===================================
+     ☑ : Test that the network speed is acceptable
+
+
+.. warning::
+
+    Don't assume that a Checkbox job will inherit any environment variable from
+    the parent shell, global env or any other source. There are a few exceptions
+    but in general:
+
+    - Any variable that is not in the ``environ`` section of a job is not set
+    - Any variable not declared in the ``environment`` section of a launcher or configuration file is not set
+
+If you decide to parametrize your tests using environment variables, always
+check if they are set or give them a default value via ``${...:-default}``.
+If you expect a variable to be set and it is not, always fail the test stating
+what variable you needed and what it was for. If you decide to use a default
+value, always output the value the test is going to use in the test log so that
+when you have to investigate why something went wrong, it is trivial to
+reproduce the tests with the parameters that may have made it fail.
+
+Resources
+=========
+
+Before even thinking to test if we are connected to the internet a wise
+question to ask would be: do we even have a network interface? Lets create a
+resource job to fetch this information.
```

Comment:
What is a "resource job"? Either link to doc, or explain in one sentence (good luck).

Reply:
Link to the doc and explanation is in order, I think this tutorial should assume you went through the basic tutorial and you are a yellow belt checkbox user, ready to start writing